### ISSUE TITLE
Checkbox .input event does no longer fire after value has been changed by another element's .change event

### ISSUE THREAD

**Issue by mamei16** [gradio-app/gradio#7954]
### Describe the bug

This bug occurred after updating from `gradio==3.50.*` to `gradio==4.23.*`.

In my UI I have a Checkbox, whose value can be changed by another element's `.change` listener. I only want the user to be able to trigger the Checkbox's event listener, so I use the `.input` method. After updating gradio, I noticed that this `input` listener only fired until the checkbox's value was changed for the first time by the other element's  `.change` listener. After this has happened, clicking the checkbox no longer triggers the `.input` listener.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import gradio as gr


with gr.Blocks() as demo:
    checkbox_event_text = gr.Text(label="Checkbox value after '.input' event", interactive=False)

    checkbox = gr.Checkbox("Checkbox")
    checkbox.input(fn=lambda x: x, inputs=checkbox, outputs=[checkbox_event_text])

    textbox = gr.Textbox(label='Text input. Write "True" to the check the checkbox, or anything else to uncheck it.')
    textbox.change(lambda x: True if x == "True" else False, textbox, [checkbox])

demo.launch()

```
1. Click the checkbox and see the upper output textbox change
2. Write anything in the bottom textbox
3. Click the checkbox again and notice the upper output textbox does no longer change accordingly 

### System Info

```shell
Gradio Environment Information:
------------------------------
Operating System: Linux
gradio version: 4.23.0
gradio_client version: 0.14.0

------------------------------------------------
gradio dependencies in your environment:

aiofiles: 23.2.1
altair: 5.3.0
fastapi: 0.110.1
ffmpy: 0.3.2
gradio-client==0.14.0 is not installed.
httpx: 0.27.0
huggingface-hub: 0.22.2
importlib-resources: 6.4.0
jinja2: 3.1.2
markupsafe: 2.1.3
matplotlib: 3.8.4
numpy: 1.26.4
orjson: 3.10.0
packaging: 23.2
pandas: 2.2.1
pillow: 10.3.0
pydantic: 2.6.4
pydub: 0.25.1
python-multipart: 0.0.9
pyyaml: 6.0.1
ruff: 0.3.5
semantic-version: 2.10.0
tomlkit==0.12.0 is not installed.
typer: 0.12.1
typing-extensions: 4.8.0
uvicorn: 0.29.0
authlib; extra == 'oauth' is not installed.
itsdangerous; extra == 'oauth' is not installed.


gradio_client dependencies in your environment:

fsspec: 2024.3.1
httpx: 0.27.0
huggingface-hub: 0.22.2
packaging: 23.2
typing-extensions: 4.8.0
websockets: 11.0.3
```

### Severity

Blocking usage of gradio

**Comment by abidlabs** [comment]
Hi @mamei16 I believe we may have fixed this, could you test with `gradio==4.25.0`?

**Comment by mamei16** [comment]
Unfortunately, it still occurs when I upgrade to `gradio==4.25.0`.

**Comment by mamei16** [comment]
It appears to be fixed in `gradio==4.26.0`. Thank you!
